rgw/v2/tests/s3_swift/test_object_lock.py

```diff
@@ -0,0 +1,186 @@
+                        s3_object_path = os.path.join(TEST_DATA_PATH, s3_object_name)
+                        log.info(f"s3 object path: {s3_object_path}")
+                        log.info("upload type: normal")
+                        io_generator(TEST_DATA_PATH + "/" + s3_object_name, size)
+                        s3_conn_client.put_object(
+                            Body=TEST_DATA_PATH + "/" + s3_object_name,
+                            Bucket=bucket_name_to_create,
+                            Key=s3_object_name,
+                        )
+                    log.info("Verify version count")
+                    # Verify version count
+                    versions = s3_conn_client.list_object_versions(
+                        Bucket=bucket_name_to_create
+                    )
+                    versions_count = len(versions["Versions"])
+                    error_message = (
+                        f"Expected: {config.objects_count}, Actual: {versions_count}"
+                    )
+                    assert versions_count == config.objects_count, error_message
+                    # Verify delete disabled for object
+                    log.info("Verify delete disabled for object")
+                    for version_dict in versions["Versions"]:
+                        try:
+                            s3_conn_client.delete_object(
+                                Bucket=bucket_name_to_create,
+                                Key=s3_object_name,
+                                VersionId=version_dict["VersionId"],
+                            )
+                            raise AccessDeniedObjectDeleted(
+                                "Access denied object deleted"
+                            )
+                        except boto3exception.ClientError as e:
+                            expected_code = "AccessDenied"
+                            actual_code = e.response["Error"]["Code"]
+                            assert (
+                                actual_code == expected_code
+                            ), "Expected: {expected_code}, Actual: {actual_code}"
+
+    # check for any crashes during the execution
+    crash_info = reusable.check_for_crash()
+    if crash_info:
+        raise TestExecError("ceph daemon crash found!")
+
+
+if __name__ == "__main__":
+
+    test_info = AddTestInfo("create m buckets with n objects")
+    test_info.started_info()
+
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        ceph_conf = CephConfOp()
+        rgw_service = RGWService()
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info("TEST_DATA_PATH: {TEST_DATA_PATH}")
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(description="Object Lock Configuration")
```

Comment:
Can we move to `docopt`? You have put in a very nice module description...

Reply:
Updated the description with usage. But not used docopt as the function exits when I use docopt.

Reply:
docopt is a good idea. i was meaning to get most of the code to use docopt. we can definitely make docopt usage here. maybe a next PR with all the changes.